### PR TITLE
AspectJ implementation with teswiz

### PR DESCRIPTION
**Pull request by MedhaGupta-Ril**
This PR covers the implementation of AspectJ with Teswiz framework
- Added new dependency and plugin for AspectJ.
- Added AspectJ aspect to add auto-generated loggers when the BL or screen methods are been initiated or completed execution.
- Since the scope of the aspect could differ according to the client's requirement, keep the aspect file within the test scope in the aspect package and kept the implementation within the main so that we have a common  Logger message.

### CODE REVIEW

File: src/test/java/com/znsio/teswiz/aspect/AspectLogging.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        long threadId = Thread.currentThread().getId();
+        context = SessionContext.getTestExecutionContext(threadId);
+    }
+    @Pointcut("execution(public * com.znsio.teswiz.businessLayer.*.*.*(..)) || execution(public * com.znsio.teswiz.screen.*.*.*.*(..))")
+    public void executionScope(){
+    }
+
+    @Before("executionScope()")
+    public void beforeAnyMethod(JoinPoint joinPoint) {
+        new AspectJMethodLoggers().beforeAnyMethod(joinPoint);
```
Comment (anandbagmar) [inline src/test/java/com/znsio/teswiz/aspect/AspectLogging.java:25]: Can you make these static methods? `AspectJLogger().beforeMethod(joinPoint)` and `AspectJLogger().afterMethod(joinPoint)`

File: src/test/java/com/znsio/teswiz/aspect/AspectLogging.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@Aspect
+public class AspectLogging {
+    private static final Logger LOGGER = Logger.getLogger(AspectLogging.class.getName());
+    private final TestExecutionContext context;
+
+    public AspectLogging() {
+        long threadId = Thread.currentThread().getId();
+        context = SessionContext.getTestExecutionContext(threadId);
+    }
+    @Pointcut("execution(public * com.znsio.teswiz.businessLayer.*.*.*(..)) || execution(public * com.znsio.teswiz.screen.*.*.*.*(..))")
```
Comment (anandbagmar) [inline src/test/java/com/znsio/teswiz/aspect/AspectLogging.java:19]: Have you checked with additional packages as well? ex: Screens, entities, etc.

File: src/test/java/com/znsio/teswiz/aspect/AspectLogging.java (hunk 1 above)
Comment (MedhaGupta-Ril) [reply src/test/java/com/znsio/teswiz/aspect/AspectLogging.java:25]: Updated

File: src/main/java/com/znsio/teswiz/aspects/AspectJMethodLoggers.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    private static final Logger LOGGER = Logger.getLogger(AspectJMethodLoggers.class.getName());
+    private final TestExecutionContext context;
+
+    public AspectJMethodLoggers() {
+        long threadId = Thread.currentThread().getId();
+        context = SessionContext.getTestExecutionContext(threadId);
+    }
+
+    public static void beforeAnyMethod(JoinPoint joinPoint) {
+        LOGGER.info(String.format("Entering method: %s ", joinPoint.getSignature().getName()));
```
Comment (anandbagmar) [inline src/main/java/com/znsio/teswiz/aspects/AspectJMethodLoggers.java:18]: This approach still will result in 2 log messages when there are parameters. There should be only 1 log message per method

Comment (MedhaGupta-Ril) [reply src/main/java/com/znsio/teswiz/aspects/AspectJMethodLoggers.java]: Updated

**Pull request merged**